### ISSUE TITLE
Question about standalone deployment on CentOS 7.6

### ISSUE THREAD

**Issue by lymanblue** [opencv/dldt#146]
Hi~

We are planning to adopt openvino for participating NIST FRVT competition.
The competition only provide a clean CentOS 7.6 environment, and the additional installation may not be allowed.
Therefore, we want to know what the libraries are required for a standalone deployment of openvino or the inference engine.

Thank you.

**Comment by shubha-ramani** [comment]
Dear @lymanblue 

With CentOS 7.6 you should have no problem.
Make sure to cd into the install_depencencies folder and run the script _install_all_dependencies.sh . Also avoid installing as root, but if you must install as root you may have to make changes to your /etc/sudoers file (Typically the "Defaults" line).  

All the instructions are here:
[https://docs.openvinotoolkit.org/latest/_docs_install_guides_installing_openvino_linux.html](https://docs.openvinotoolkit.org/latest/_docs_install_guides_installing_openvino_linux.html)

If you follow the instructions carefully you should have no difficulties installing on CentOS 7.6 .

Thanks,

Shubha

**Comment by lymanblue** [comment]
Hi~

If we cannot install OpenVINO on the target system.
And the competition request us to submit binary files (e.g., so files).
Are these instructions still applicable?

We can build the suggested environment on our own server.
If we want to build a dynamic library (e.g., so files) to attend the competition.
What are the required files/libraries to be included in our submission folder?
How to make a standalone deployment on CentOS in this situation?

**Comment by alalek** [comment]
> what the libraries are required for a standalone deployment of openvino

Recommended way is using installer.
Including acceptance of EULA.

---

You can try to build DLDT from this repository in your own way (I believe support of CPU H/W is enough).

--

> .so files

`LD_DEBUG=files your_app` is a great tool to capture that information

**Comment by lymanblue** [comment]
I see. Thank you very much.

**Comment by lymanblue** [comment]
Hi~ We didn't adopt openvino as our solution in the end since additional software installation is not allowed.
Thank you for your suggestion.

**Comment by antonrom23** [comment]
@lymanblue , sorry about your decision. Can we close this issue?

**Comment by lymanblue** [comment]
ok, thank you.

**Comment by ngxbac** [comment]
@antonrom23 , @shubha-ramani 

I found that there are some works (at least 2 teams), they used OpenVINO for the NIST FRVT competition.
One of them reported that they modified the internal code of OpenVINO to match the rules of this competition. That is, you are only allowed to run in a single thread. However, in default, OpenVINO uses at least 3 threads.

As mentioned in here #185 , 
```
two aditional threads are required to support asynchronious mode and callbacks. They are created everytime for each executable network.
```
Based on these comments, I found that two single threads are: [CallbackExcecutor](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/mkldnn_exec_network.cpp#L115-L120) and [TaskExecutor](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/mkldnn_exec_network.cpp#L113). 

I think by commenting CallbackExcecutor, I could reduce one thread. 
For the TaskExecutor, it is quite different. I tried to understand the OpenVINO code and design. It seems to me that, this executor runs the network graph and wait for the InferenceRequest. 
I wonder if we have another way to infer without using this thread?. 

Many thanks.
